client/src/app/queries/businessservices.ts

```diff
@@ -1,45 +1,75 @@
 import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { AxiosError } from "axios";
 
 import {
+  createBusinessService,
   deleteBusinessService,
   getBusinessServiceById,
   getBusinessServices,
+  updateBusinessService,
 } from "@app/api/rest";
-import { BusinessService } from "@app/api/models";
-import { AxiosError } from "axios";
 
 export const BusinessServicesQueryKey = "businessservices";
 export const BusinessServiceQueryKey = "businessservice";
 
 export const useFetchBusinessServices = () => {
-  const { data, isLoading, error, refetch } = useQuery<BusinessService[]>(
-    [BusinessServicesQueryKey],
-    async () => (await getBusinessServices()).data,
-    {
-      onError: (error) => console.log("error, ", error),
-    }
-  );
+  const { data, isLoading, error, refetch } = useQuery({
+    queryKey: [BusinessServicesQueryKey],
+    queryFn: getBusinessServices,
+    onError: (error: AxiosError) => console.log("error, ", error),
+  });
   return {
     businessServices: data || [],
     isFetching: isLoading,
     fetchError: error as AxiosError,
     refetch,
   };
 };
-export const useFetchBusinessServiceByID = (id: number | string) => {
-  const { data, isLoading, error } = useQuery<BusinessService>(
-    [BusinessServicesQueryKey, id],
-    async () => (await getBusinessServiceById(id)).data,
-    { onError: (error) => console.log(error) }
-  );
 
+export const useFetchBusinessServiceByID = (id: number | string) => {
+  const { data, isLoading, error } = useQuery({
+    queryKey: [BusinessServicesQueryKey, id],
+    queryFn: () => getBusinessServiceById(id),
+    onError: (error: AxiosError) => console.log("error, ", error),
+  });
   return {
     businessService: data,
     isFetching: isLoading,
     fetchError: error as AxiosError,
   };
 };
 
+export const useCreateBusinessServiceMutation = (
+  onSuccess: (res: any) => void,
+  onError: (err: AxiosError) => void
+) => {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: createBusinessService,
+    onSuccess: (res) => {
+      onSuccess(res);
+      queryClient.invalidateQueries([BusinessServicesQueryKey]);
+    },
+    onError,
+  });
+};
+
+export const useUpdateBusinessServiceMutation = (
+  onSuccess: (res: any) => void,
+  onError: (err: AxiosError) => void
+) => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: updateBusinessService,
+    onSuccess: (res) => {
+      onSuccess(res);
```

Comment:
```suggestion
  onSuccess: (_, vars) => {
      onSuccess(vars.businessService.name);
```

Reply:
@ibolton336,

This seems to be out of scope, please see why here https://github.com/konveyor/tackle2-ui/issues/1214 to following-up.

Reply:
I don't mind not doing this. Just seemed like an easy solution that would help standardize notifications with more info.

Reply:
@ibolton336, I agree but I would like to have an agreed overall approach to address this across the board.